flink-connectors/flink-jdbc/src/main/java/org/apache/flink/api/java/io/jdbc/JDBCOutputFormat.java

```diff
@@ -87,8 +89,12 @@ public void open(int taskNumber, int numTasks) throws IOException {
 	}
 
 	@Override
-	public void writeRecord(Row row) {
+	public void writeRecord(Row row) throws IOException{
 		try {
+			if (!connection.isValid(connectionCheckTimeout)) {
+				LOG.error("JDBC connection is closed,start to open a new connection");
+				establishConnectionAndPreparedStatement();
```

Comment:
Maybe you should close the connection and statement fisrt, and then reestablish.